web/shared/src/internal_url.ts

```diff
@@ -50,20 +50,36 @@ export function encode_stream_id(
     return encodeHashComponent(slug);
 }
 
+export function maybe_encode_with_operator(
+    channel_topic_url: string,
+    message_id: number | undefined,
+): string {
+    if (message_id === undefined) {
+        return channel_topic_url;
+    }
+
+    return channel_topic_url + `/with/${message_id}`;
+}
```

Comment:
Do you expect to use this function elsewhere? If not, it'd probably be nicer to just have the caller construct two types of URLs in an if/else statement, so you can see the whole construction in one place.

Reply:
Inlined it.